src/components/NewCommentForm.tsx

```diff
@@ -60,7 +64,12 @@ export const NewCommentForm: React.FC<Props> = ({ onSubmit }) => {
     setSubmitting(true);
 
     // it is very easy to forget about `await` keyword
-    await onSubmit({ name, email, body });
+    await addComment({
+      name,
+      email,
+      body,
+      postId: selectedPost!.id,
```

Comment:
The use of `selectedPost!` assumes that `selectedPost` is never null or undefined. This could lead to runtime errors if `selectedPost` is not set. Consider adding a check to ensure `selectedPost` is defined before accessing its properties.